### DISCUSSION TITLE
Ontology for ephys data?

### DISCUSSION THREAD

**Discussion by stevejryan** [NeurodataWithoutBorders/helpdesk#90] [General]
Hey folks,

I'm wondering if any NWB users are aware of an ontology or controlled vocabulary that describes features that can be extracted from intracellular voltage measurements? I'm thinking of something that would contain concepts / terms like action potential, spike train, burst, spike frequency adaptation, latency, inter-spike interval, spike width, rise-time, that sort of thing.

I haven't found anything online or in the literature, but before I give up I thought I'd check around the water cooler and see if anyone knows anything :) 

Honestly even examples of ontologies describing extracted features from other single-cell-resolution assays would be interesting as a starting place.

Thanks!

**Comment by oruebel** [comment]
I am not aware of single specific ontology but I think looking at [BioPortal ](https://bioportal.bioontology.org/search) might be useful to find potential related ontologies. For example:

- **[action potential](https://bioportal.bioontology.org/search?q=action%20potential&ontologies=&include_properties=false&include_views=false&includeObsolete=false&require_definition=false&exact_match=false&categories=)** has some potentially useful definitions in:
    - [Gene Ontology](https://bioportal.bioontology.org/ontologies/GO) : https://bioportal.bioontology.org/ontologies/GO?p=classes&conceptid=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FGO_0001508  
    - [Computer Retrieval of Information on Scientific Projects Thesaurus (CRISP)](https://bioportal.bioontology.org/ontologies/CRISP): https://bioportal.bioontology.org/ontologies/CRISP?p=classes&conceptid=http%3A%2F%2Fpurl.bioontology.org%2Fontology%2FCSP%2F1012-4683
-  **[spike train](https://bioportal.bioontology.org/search?q=spike%20train&ontologies=&include_properties=false&include_views=false&includeObsolete=false&require_definition=false&exact_match=false&categories=)** is also defined in the [Gene Ontology](https://bioportal.bioontology.org/ontologies/GO)  https://bioportal.bioontology.org/ontologies/GO?p=classes&conceptid=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FGO_0098874

So maybe the  [Gene Ontology (GO)](https://bioportal.bioontology.org/ontologies/GO) could be start? Even if a single ontology may not have all the terms, you could look at compiling your own list of terms in a TermSet. In the context of NWB we are looking at using LinkML to define sets of terms. I believe these are the relevant tutorials in that regard for NWB:

- Demo notebook from the Brain Data Modeling Workshop: https://drive.google.com/drive/folders/1W55-IByAKW1dWDpSHOpx0CKtOCNzjSvi?usp=sharing 
- https://pynwb.readthedocs.io/en/latest/tutorials/general/plot_configurator.html#sphx-glr-tutorials-general-plot-configurator-py
- https://hdmf.readthedocs.io/en/stable/tutorials/plot_term_set.html#sphx-glr-tutorials-plot-term-set-py

Another effort that may be of interest for you is the https://incf.org/sig/incf-working-group-electrophysiology-stimulation-ontology This working group was looking at defining an ontology for describing the electrical stimuli that are being using in intracellular electrophysiology.
